build

```diff
@@ -1,24 +1,26 @@
 #!/bin/bash -e
 
+if [ ! -z ${SHALLOW_CLONE} ]; then
+	# Can't use --shallow-submodules because some bug in either Git or in the
+	# mupdf submodule setup makes the clone command fail.
+	SHALLOW_CLONE="--depth 1 --no-shallow-submodules"
+fi
+
 MUPDF_VERSION="1.13.0"
 
 if [[ ! -d mupdf ]]; then
 	echo "Downloading muPDF..."
-	git clone https://github.com/ArtifexSoftware/mupdf.git
-	( cd mupdf && git checkout -q ${MUPDF_VERSION} )
-	( cd mupdf/thirdparty && git submodule init && git submodule update --recursive )
+	# Setting -j0 clones submodules with as many parallel jobs as possible
+	git clone --branch ${MUPDF_VERSION} ${SHALLOW_CLONE} --recurse-submodules=thirdparty -j0 https://github.com/ArtifexSoftware/mupdf.git
 fi
 
 echo "Building muPDF..."
-cd mupdf
-
-XCFLAGS="-g" make -j8 libs
+(cd mupdf; XCFLAGS="-g" make -j8 libs)
```

Comment:
Agreed, this shouldn't be hardcoded if it runs on different systems. You can use instead `make -j $(nproc)`. [nproc(1)](https://linux.die.net/man/1/nproc) is a Linux command which read `/proc/cpuinfo` to output the number of available CPU cores.

Reply:
Or `-j $(($(nproc) + 1))` to make sure there will no be any problem when it's I/O bound.